src/traces/parcats/attributes.js

```diff
@@ -0,0 +1,193 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var extendFlat = require('../../lib/extend').extendFlat;
+var plotAttrs = require('../../plots/attributes');
+var fontAttrs = require('../../plots/font_attributes');
+var colorAttributes = require('../../components/colorscale/attributes');
+var domainAttrs = require('../../plots/domain').attributes;
+var scatterAttrs = require('../scatter/attributes');
+var scatterLineAttrs = scatterAttrs.line;
+var colorbarAttrs = require('../../components/colorbar/attributes');
+
+var line = extendFlat({
+    editType: 'calc'
+}, colorAttributes('line', {editType: 'calc'}),
+    {
+        showscale: scatterLineAttrs.showscale,
+        colorbar: colorbarAttrs,
+        shape: {
+            valType: 'enumerated',
+            values: ['linear', 'hspline'],
+            dflt: 'linear',
+            role: 'info',
+            editType: 'plot',
+            description: [
+                'Sets the shape of the paths.',
+                'If `linear`, paths are composed of straight lines.',
+                'If `hspline`, paths are composed of horizontal curved splines'
+            ].join(' ')
+        }
+    });
+
+module.exports = {
+    domain: domainAttrs({name: 'parcats', trace: true, editType: 'calc'}),
+    hoverinfo: extendFlat({}, plotAttrs.hoverinfo, {
+        flags: ['count', 'probability'],
+        editType: 'plot'
+        // plotAttrs.hoverinfo description is appropriate
+    }),
+    hovermode: {
```

Comment:
A little late to the party, but I'd say having `hovermode` as a layout and a trace may sounds very confusing. 

It would be nice to make whether attribute name we pick here also describe the "separate vs combined" cases for finance trace hover labels (cc https://github.com/plotly/plotly.js/pull/2959).

Reply:
I just ready through #2959 and I agree it sounds like we could use a new property to specify different approaches to displaying tooltips.  Some ideas based on synonyms for "mode":
 - `hoverbehavior`
 - `hoverform`
 - `hoverkind`
 - `hovermethod`
 - `hovertype`
 - `hovervariety`

Any of those sound reasonable? cc @alexcjohnson

Reply:
Sure thing!